plugin/src/main/java/org/gradle/testretry/internal/TestTaskConfigurer.java

```diff
@@ -90,12 +91,34 @@ private static boolean supportsPropertyConventions(VersionNumber gradleVersion)
         return gradleVersion.getMajor() == 5 ? gradleVersion.getMinor() >= 1 : gradleVersion.getMajor() > 5;
     }
 
-    private static class FinalizeTaskAction implements Action<Task> {
+    private static class ConditionalTaskAction implements Action<Task> {
+
+        private static final String DEACTIVATION_PROPERTY_NAME = "retry.deactivated";
+
+        private final Action<Test> delegate;
+
+        public ConditionalTaskAction(Action<Test> delegate) {
+            this.delegate = delegate;
+        }
 
         @Override
         public void execute(@NotNull Task task) {
-            Test testTask = (Test) task;
-            TestExecuter<JvmTestExecutionSpec> testExecuter = getTestExecuter(testTask);
+            ExtraPropertiesExtension extra = task.getExtensions().getExtraProperties();
+            boolean deactivated = extra.has(DEACTIVATION_PROPERTY_NAME) && Boolean.parseBoolean(String.valueOf(extra.get(DEACTIVATION_PROPERTY_NAME)));
```

Comment:
What's this about? I don't see any mention of it in the spec, and don't understand what it's for.

Reply:
Resolved in other channels.